### PR TITLE
upgrade remark to version 8 and unified to version 9

### PR DESCRIPTION
**Pull request by ChristianMurphy**
These versions bring improvements to commonmark compliance and official typings.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -5,65 +5,22 @@
...
 // - ClassicDarkChocolate <https://github.com/ClassicDarkChocolate>
 // - Espen Hovlandsdal <https://espen.codes/>
+// - Christian Murphy <https://github.com/ChristianMurphy>
 
-import {Component, ReactElement, ReactNode, ReactType} from 'react'
-
-declare class ReactMarkdown extends Component<ReactMarkdown.ReactMarkdownProps, {}> {}
+import {Component, ReactElement, ReactNode} from 'react'
+import {PluggableList} from 'unified'
+import {Content} from 'mdast'
```
Comment (ChristianMurphy) [inline index.d.ts:12]: Pulls in official types and leverages them

File: test/react-markdown.test.js [hunk 2]
```diff
@@ -224,7 +224,7 @@ test('should handle ordered lists with a start index', () => {
 })
 
 test('should pass depth, index and ordered props to list/listItem', () => {
-  const input = '- foo\n  2. bar\n  3. baz\n- root\n'
+  const input = '- foo\n\n  2. bar\n  3. baz\n\n- root\n'
```
Comment (ChristianMurphy) [inline test/react-markdown.test.js:227]: This has been updated to conform with commonmark, the previous output was a bug.

**Comment by gilisho** [comment]
Can this be merged? 🤞 
I came across [this](https://github.com/rexxars/react-markdown/issues/235) issue, and apparently   the solution is to really bump the version of `remark-parser`.

**Comment by starpit** [comment]
we'd also like to see this merged. anything we can do to help?

**Comment by gilisho** [comment]
@rexxars sorry for tagging, but I would be really happy if this could be merged, so I'm just pinging.

**Comment by wooorm** [comment]
@ChristianMurphy I can merge this now, I believe?

**Comment by ChristianMurphy** [comment]
Yep, this should be good to go

**Comment by wooorm** [comment]
✨

**Pull request merged**